modules/Core/classes/Middleware/Global/TFAMiddleware.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Two-Factor Authentication middleware hook.
+ * Enforces TFA requirements for users in groups that require it.
+ *
+ * @package NamelessMC\Hooks
+ * @author Aberdeener
+ * @version 2.3.0
+ * @license MIT
+ */
+class TFAMiddleware extends AbstractMiddleware
+{
+    public array $exemptRoutes = [
+        '/logout',
+        '/user/settings' // Allow access to settings to enable TFA
+    ];
+
+    public function handle(User $user, Request $request, Language $language): void
+    {
+        // Only process for logged-in users
+        if (!$user->isLoggedIn()) {
+            return;
+        }
+
+        // Skip if AJAX request, such as Alert or PM checks
```

Comment:
Was well also to prevent the code from refreshing when user switch to next step as that refreshing the site